File: tags/NHI1-0.12/theLink/example/ruby/Filter3.rb
#+
#§  \file       theLink/example/ruby/Filter3.rb
#§  \brief      \$Id: LbMain 26 2009-12-03 11:48:43Z aotto1968 $
#§  
#§  (C) 2010 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev: 26 $
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

require "rubymsgque"

class Filter3 < MqS
  def initialize(tmpl = nil)
    super(tmpl)
    ConfigSetServerSetup(method(:ServerSetup))
  end
  def ServerSetup
    ftr = ServiceGetFilter()
    ServiceProxy("+ALL")
    ServiceProxy("+TRT")
    ftr.ServiceProxy("+ALL")
    ftr.ServiceProxy("+TRT")
  end
end
srv = FactoryNew(Filter3)
begin
  srv.LinkCreate($0,ARGV)
  srv.ProcessEvent(MqS::WAIT_FOREVER)
rescue SignalException => ex
  # ignore
rescue Exception => ex
  srv.ErrorSet(ex)
ensure
  srv.Exit()
end

File: tags/NHI1-0.12/theLink/example/php/Filter3.php
<?php
#+
#§  \file       theLink/example/php/Filter3.php
#§  \brief      \$Id$
#§  
#§  (C) 2010 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev$
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

class Filter3 extends MqS implements iServerSetup {
  public function ServerSetup() {
    $ftr = $this->ServiceGetFilter();
    $this->ServiceProxy("+ALL");
    $this->ServiceProxy("+TRT");
    $ftr->ServiceProxy("+ALL");
    $ftr->ServiceProxy("+TRT");
  }
}
$srv = FactoryNew('Filter3');
try {
  $srv->LinkCreate($argv);
  $srv->ProcessEvent(MqS::WAIT_FOREVER);
} catch (Exception $ex) {
  $srv->ErrorSet($ex);
}
$srv->Exit();

?>
